[source-hubspot] - Handle custom objects where name collides with standard objects

jtruty opened the pull request:
## What
This handles custom object names that collide with standard object names. Currently if this is the case, stream discovery fails.
Fixes: https://github.com/airbytehq/airbyte/issues/40679

I re-opened this PR from a non-org github account to allow maintainers to update the PR. This one: https://github.com/airbytehq/airbyte/pull/41639 is the original and will be closed.

## How
We check the custom object names against a set of standard hubspot object names. If there is a collision, we use the fully qualified name for the custom object instead of the actual object name.

## Review guide
airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

## User Impact
Stream discovery should work when this happens and allow a normal sync to work.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚
- [ ] NO ❌

jtruty committed:
Add support for Hubspot Deal Splits

jtruty commented:
@marcosmarxm I opened this from a non-org GH to allow maintainer edits.

marcosmarxm commented on docs/integrations/sources/hubspot.md:
```diff
@@ -337,6 +337,7 @@ The connector is restricted by normal HubSpot [rate limitations](https://legacyd
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                                                                          |
 |:--------|:-----------| :------------------------------------------------------- |:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 4.2.13 | 2024-07-18 | [41639](https://github.com/airbytehq/airbyte/pull/41639) | Handle custom object name collisions with standard objects |
```
It is missing the metadata.yaml and pyproject connector version update.  Please fix that.

jtruty replied:
@marcosmarxm updated!

jtruty commented:
@marcosmarxm @airbytehq/dev-python any more thoughts on this?

jtruty commented:
@marcosmarxm the files are updated and conflicts fixed - anything else I can do here?

alafanechere committed:
🐙 source-insightly: run up-to-date pipeline [2024-07-27] (#42785)

alafanechere committed:
🐙 source-pivotal-tracker: run up-to-date pipeline [2024-07-27] (#42784)

alafanechere committed:
🐙 source-newsdata: run up-to-date pipeline [2024-07-27] (#42783)

alafanechere committed:
🐙 source-linkedin-pages: run up-to-date pipeline [2024-07-27] (#42782)

alafanechere committed:
🐙 destination-cumulio: run up-to-date pipeline [2024-07-27] (#42780)

alafanechere committed:
🐙 source-bamboo-hr: run up-to-date pipeline [2024-07-27] (#42779)

alafanechere committed:
🐙 source-tempo: run up-to-date pipeline [2024-07-27] (#42778)

alafanechere committed:
🐙 source-smaily: run up-to-date pipeline [2024-07-27] (#42777)

alafanechere committed:
🐙 source-datadog: run up-to-date pipeline [2024-07-27] (#42776)

alafanechere committed:
🐙 source-pocket: run up-to-date pipeline [2024-07-27] (#42775)

alafanechere committed:
🐙 source-google-webfonts: run up-to-date pipeline [2024-07-27] (#42774)

alafanechere committed:
🐙 source-bigcommerce: run up-to-date pipeline [2024-07-27] (#42773)

alafanechere committed:
🐙 source-workable: run up-to-date pipeline [2024-07-27] (#42772)

alafanechere committed:
🐙 source-polygon-stock-api: run up-to-date pipeline [2024-07-27] (#42771)

alafanechere committed:
🐙 source-google-pagespeed-insights: run up-to-date pipeline [2024-07-27] (#42770)

alafanechere committed:
🐙 source-posthog: run up-to-date pipeline [2024-07-27] (#42769)

alafanechere committed:
🐙 source-lemlist: run up-to-date pipeline [2024-07-27] (#42768)

alafanechere committed:
🐙 source-pexels-api: run up-to-date pipeline [2024-07-27] (#42767)

alafanechere committed:
🐙 source-coinmarketcap: run up-to-date pipeline [2024-07-27] (#42766)

alafanechere committed:
🐙 source-chargify: run up-to-date pipeline [2024-07-27] (#42765)

alafanechere committed:
🐙 source-senseforce: run up-to-date pipeline [2024-07-27] (#42763)

alafanechere committed:
🐙 source-waiteraid: run up-to-date pipeline [2024-07-27] (#42762)

alafanechere committed:
🐙 destination-timeplus: run up-to-date pipeline [2024-07-27] (#42761)

alafanechere committed:
🐙 source-todoist: run up-to-date pipeline [2024-07-27] (#42760)

alafanechere committed:
🐙 destination-rabbitmq: run up-to-date pipeline [2024-07-27] (#42759)

alafanechere committed:
🐙 source-mailerlite: run up-to-date pipeline [2024-07-27] (#42758)

alafanechere committed:
🐙 source-sap-fieldglass: run up-to-date pipeline [2024-07-27] (#42756)

alafanechere committed:
🐙 source-toggl: run up-to-date pipeline [2024-07-27] (#42755)

alafanechere committed:
🐙 source-outbrain-amplify: run up-to-date pipeline [2024-07-27] (#42754)

alafanechere committed:
🐙 destination-duckdb: run up-to-date pipeline [2024-07-27] (#42753)

alafanechere committed:
🐙 source-surveymonkey: run up-to-date pipeline [2024-07-27] (#42752)

alafanechere committed:
🐙 source-convex: run up-to-date pipeline [2024-07-27] (#42751)

alafanechere committed:
🐙 source-shortio: run up-to-date pipeline [2024-07-27] (#42750)

alafanechere committed:
🐙 source-vantage: run up-to-date pipeline [2024-07-27] (#42749)

alafanechere committed:
🐙 source-commcare: run up-to-date pipeline [2024-07-27] (#42748)

alafanechere committed:
🐙 source-mailjet-sms: run up-to-date pipeline [2024-07-27] (#42747)

alafanechere committed:
🐙 source-google-analytics-data-api: run up-to-date pipeline [2024-07-27] (#42746)

alafanechere committed:
🐙 source-freshsales: run up-to-date pipeline [2024-07-27] (#42744)

alafanechere committed:
🐙 source-aircall: run up-to-date pipeline [2024-07-27] (#42743)

alafanechere committed:
🐙 source-coda: run up-to-date pipeline [2024-07-27] (#42741)

alafanechere committed:
🐙 source-sendinblue: run up-to-date pipeline [2024-07-27] (#42740)

alafanechere committed:
🐙 source-okta: run up-to-date pipeline [2024-07-27] (#42739)

alafanechere committed:
🐙 source-pokeapi: run up-to-date pipeline [2024-07-27] (#42738)

alafanechere committed:
🐙 source-aha: run up-to-date pipeline [2024-07-27] (#42737)

alafanechere committed:
🐙 source-kyve: run up-to-date pipeline [2024-07-27] (#42736)

alafanechere committed:
🐙 source-harness: run up-to-date pipeline [2024-07-27] (#42735)

alafanechere committed:
🐙 source-azure-table: run up-to-date pipeline [2024-07-27] (#42734)

alafanechere committed:
🐙 source-nytimes: run up-to-date pipeline [2024-07-27] (#42733)

alafanechere committed:
🐙 source-gainsight-px: run up-to-date pipeline [2024-07-27] (#42732)

alafanechere committed:
🐙 source-openweather: run up-to-date pipeline [2024-07-27] (#42731)

alafanechere committed:
🐙 source-qualaroo: run up-to-date pipeline [2024-07-27] (#42730)

alafanechere committed:
🐙 source-sendgrid: run up-to-date pipeline [2024-07-27] (#42729)

alafanechere committed:
🐙 source-linkedin-ads: run up-to-date pipeline [2024-07-27] (#42728)

alafanechere committed:
🐙 source-getlago: run up-to-date pipeline [2024-07-27] (#42727)

alafanechere committed:
🐙 source-typeform: run up-to-date pipeline [2024-07-27] (#42726)

alafanechere committed:
🐙 source-tplcentral: run up-to-date pipeline [2024-07-27] (#42725)

alafanechere committed:
🐙 source-instatus: run up-to-date pipeline [2024-07-27] (#42724)

alafanechere committed:
🐙 source-recharge: run up-to-date pipeline [2024-07-27] (#42723)

alafanechere committed:
🐙 source-fastbill: run up-to-date pipeline [2024-07-27] (#42722)

alafanechere committed:
🐙 source-instagram: run up-to-date pipeline [2024-07-27] (#42721)

alafanechere committed:
🐙 source-zapier-supported-storage: run up-to-date pipeline [2024-07-27] (#42720)

alafanechere committed:
🐙 source-declarative-manifest: run up-to-date pipeline [2024-07-27] (#42719)

alafanechere committed:
🐙 source-twilio-taskrouter: run up-to-date pipeline [2024-07-27] (#42718)

alafanechere committed:
🐙 source-captain-data: run up-to-date pipeline [2024-07-27] (#42717)

alafanechere committed:
🐙 source-mailgun: run up-to-date pipeline [2024-07-27] (#42716)

alafanechere committed:
🐙 source-dockerhub: run up-to-date pipeline [2024-07-27] (#42715)

alafanechere committed:
🐙 source-google-ads: run up-to-date pipeline [2024-07-27] (#42714)

alafanechere committed:
🐙 source-pagerduty: run up-to-date pipeline [2024-07-27] (#42713)

alafanechere committed:
🐙 source-younium: run up-to-date pipeline [2024-07-27] (#42712)

alafanechere committed:
🐙 source-intruder: run up-to-date pipeline [2024-07-27] (#42711)

alafanechere committed:
🐙 source-copper: run up-to-date pipeline [2024-07-27] (#42710)

alafanechere committed:
🐙 source-recreation: run up-to-date pipeline [2024-07-27] (#42709)

alafanechere committed:
🐙 source-retently: run up-to-date pipeline [2024-07-27] (#42708)

alafanechere committed:
🐙 source-rki-covid: run up-to-date pipeline [2024-07-27] (#42707)

alafanechere committed:
🐙 source-wikipedia-pageviews: run up-to-date pipeline [2024-07-27] (#42706)

alafanechere committed:
🐙 destination-vectara: run up-to-date pipeline [2024-07-27] (#42705)

alafanechere committed:
🐙 source-microsoft-sharepoint: run up-to-date pipeline [2024-07-27] (#42704)

alafanechere committed:
🐙 destination-firebolt: run up-to-date pipeline [2024-07-27] (#42703)

alafanechere committed:
🐙 source-configcat: run up-to-date pipeline [2024-07-27] (#42702)

alafanechere committed:
🐙 source-gutendex: run up-to-date pipeline [2024-07-27] (#42701)

alafanechere committed:
🐙 source-mailersend: run up-to-date pipeline [2024-07-27] (#42700)

alafanechere committed:
🐙 source-confluence: run up-to-date pipeline [2024-07-27] (#42699)

alafanechere committed:
🐙 source-adjust: run up-to-date pipeline [2024-07-27] (#42698)

alafanechere committed:
🐙 source-serpstat: run up-to-date pipeline [2024-07-27] (#42697)

alafanechere committed:
🐙 source-everhour: run up-to-date pipeline [2024-07-27] (#42696)

alafanechere committed:
🐙 source-fleetio: run up-to-date pipeline [2024-07-27] (#42695)

alafanechere committed:
🐙 source-fullstory: run up-to-date pipeline [2024-07-27] (#42694)

alafanechere committed:
🐙 source-gcs: run up-to-date pipeline [2024-07-27] (#42693)

alafanechere committed:
🐙 source-hubplanner: run up-to-date pipeline [2024-07-27] (#42692)

alafanechere committed:
🐙 source-microsoft-teams: run up-to-date pipeline [2024-07-27] (#42691)

alafanechere committed:
🐙 source-smartengage: run up-to-date pipeline [2024-07-27] (#42690)

alafanechere committed:
🐙 source-tyntec-sms: run up-to-date pipeline [2024-07-27] (#42689)

alafanechere committed:
🐙 source-hubspot: run up-to-date pipeline [2024-07-27] (#42688)

alafanechere committed:
🐙 source-rss: run up-to-date pipeline [2024-07-27] (#42687)

alafanechere committed:
🐙 source-plaid: run up-to-date pipeline [2024-07-27] (#42686)

alafanechere committed:
🐙 source-salesforce: run up-to-date pipeline [2024-07-27] (#42685)

alafanechere committed:
🐙 destination-snowflake-cortex: run up-to-date pipeline [2024-07-27] (#42684)

alafanechere committed:
🐙 source-mailchimp: run up-to-date pipeline [2024-07-27] (#42683)

alafanechere committed:
🐙 source-faker: run up-to-date pipeline [2024-07-27] (#42682)

alafanechere committed:
🐙 source-appsflyer: run up-to-date pipeline [2024-07-27] (#42681)

alafanechere committed:
🐙 source-snapchat-marketing: run up-to-date pipeline [2024-07-27] (#42680)

alafanechere committed:
🐙 source-dixa: run up-to-date pipeline [2024-07-27] (#42679)

alafanechere committed:
🐙 source-hellobaton: run up-to-date pipeline [2024-07-27] (#42678)

alafanechere committed:
🐙 source-activecampaign: run up-to-date pipeline [2024-07-27] (#42677)

alafanechere committed:
🐙 source-freshcaller: run up-to-date pipeline [2024-07-27] (#42676)

alafanechere committed:
🐙 source-jina-ai-reader: run up-to-date pipeline [2024-07-27] (#42675)

alafanechere committed:
🐙 source-railz: run up-to-date pipeline [2024-07-27] (#42674)

alafanechere committed:
🐙 source-commercetools: run up-to-date pipeline [2024-07-27] (#42673)

alafanechere committed:
🐙 source-weatherstack: run up-to-date pipeline [2024-07-27] (#42672)

alafanechere committed:
🐙 source-dremio: run up-to-date pipeline [2024-07-27] (#42670)

alafanechere committed:
🐙 source-zendesk-talk: run up-to-date pipeline [2024-07-27] (#42669)

alafanechere committed:
🐙 source-zenefits: run up-to-date pipeline [2024-07-27] (#42668)

alafanechere committed:
🐙 source-zenloop: run up-to-date pipeline [2024-07-27] (#42667)

alafanechere committed:
🐙 source-quickbooks: run up-to-date pipeline [2024-07-27] (#42666)

alafanechere committed:
🐙 source-rd-station-marketing: run up-to-date pipeline [2024-07-27] (#42665)

alafanechere committed:
🐙 source-klaviyo: run up-to-date pipeline [2024-07-27] (#42664)

alafanechere committed:
🐙 source-freshservice: run up-to-date pipeline [2024-07-27] (#42663)

alafanechere committed:
🐙 source-persistiq: run up-to-date pipeline [2024-07-27] (#42661)

alafanechere committed:
🐙 source-apple-search-ads: run up-to-date pipeline [2024-07-27] (#42660)

alafanechere committed:
🐙 source-ashby: run up-to-date pipeline [2024-07-27] (#42658)

alafanechere committed:
🐙 destination-xata: run up-to-date pipeline [2024-07-27] (#42657)

alafanechere committed:
🐙 source-open-exchange-rates: run up-to-date pipeline [2024-07-27] (#42656)

alafanechere committed:
🐙 source-paystack: run up-to-date pipeline [2024-07-27] (#42655)

alafanechere committed:
🐙 source-intercom: run up-to-date pipeline [2024-07-27] (#42654)

alafanechere committed:
🐙 source-iterable: run up-to-date pipeline [2024-07-27] (#42653)

alafanechere committed:
🐙 source-zoom: run up-to-date pipeline [2024-07-27] (#42652)

alafanechere committed:
🐙 source-sonar-cloud: run up-to-date pipeline [2024-07-27] (#42651)

alafanechere committed:
🐙 source-opsgenie: run up-to-date pipeline [2024-07-27] (#42650)

alafanechere committed:
🐙 source-pardot: run up-to-date pipeline [2024-07-27] (#42649)

alafanechere committed:
🐙 source-postmarkapp: run up-to-date pipeline [2024-07-27] (#42648)

alafanechere committed:
🐙 source-clickup-api: run up-to-date pipeline [2024-07-27] (#42647)

alafanechere committed:
🐙 source-webflow: run up-to-date pipeline [2024-07-27] (#42646)

alafanechere committed:
🐙 source-merge: run up-to-date pipeline [2024-07-27] (#42645)

alafanechere committed:
🐙 source-lokalise: run up-to-date pipeline [2024-07-27] (#42644)

alafanechere committed:
🐙 source-convertkit: run up-to-date pipeline [2024-07-27] (#42643)

alafanechere committed:
🐙 source-aws-cloudtrail: run up-to-date pipeline [2024-07-27] (#42642)

alafanechere committed:
🐙 source-rocket-chat: run up-to-date pipeline [2024-07-27] (#42641)

alafanechere committed:
🐙 source-tvmaze-schedule: run up-to-date pipeline [2024-07-27] (#42640)

alafanechere committed:
🐙 source-close-com: run up-to-date pipeline [2024-07-27] (#42639)

alafanechere committed:
🐙 source-square: run up-to-date pipeline [2024-07-27] (#42638)

alafanechere committed:
🐙 source-woocommerce: run up-to-date pipeline [2024-07-27] (#42637)

alafanechere committed:
🐙 source-glassfrog: run up-to-date pipeline [2024-07-27] (#42636)

alafanechere committed:
🐙 source-timely: run up-to-date pipeline [2024-07-27] (#42635)

alafanechere committed:
🐙 source-gnews: run up-to-date pipeline [2024-07-27] (#42634)

alafanechere committed:
🐙 source-babelforce: run up-to-date pipeline [2024-07-27] (#42633)

alafanechere committed:
🐙 source-smartsheets: run up-to-date pipeline [2024-07-27] (#42632)

alafanechere committed:
🐙 source-customer-io: run up-to-date pipeline [2024-07-27] (#42631)

alafanechere committed:
🐙 source-microsoft-onedrive: run up-to-date pipeline [2024-07-27] (#42630)

alafanechere committed:
🐙 destination-weaviate: run up-to-date pipeline [2024-07-27] (#42629)

alafanechere committed:
🐙 destination-chroma: run up-to-date pipeline [2024-07-27] (#42628)

alafanechere committed:
🐙 source-apify-dataset: run up-to-date pipeline [2024-07-27] (#42627)

alafanechere committed:
🐙 source-chargebee: run up-to-date pipeline [2024-07-27] (#42626)

alafanechere committed:
🐙 source-coingecko-coins: run up-to-date pipeline [2024-07-27] (#42625)

alafanechere committed:
🐙 source-drift: run up-to-date pipeline [2024-07-27] (#42624)

alafanechere committed:
🐙 source-stripe: run up-to-date pipeline [2024-07-27] (#42623)

alafanechere committed:
🐙 source-qonto: run up-to-date pipeline [2024-07-27] (#42622)

alafanechere committed:
🐙 source-flexport: run up-to-date pipeline [2024-07-27] (#42621)

alafanechere committed:
🐙 destination-qdrant: run up-to-date pipeline [2024-07-27] (#42620)

alafanechere committed:
🐙 source-callrail: run up-to-date pipeline [2024-07-27] (#42619)

alafanechere committed:
🐙 source-amplitude: run up-to-date pipeline [2024-07-27] (#42618)

alafanechere committed:
🐙 source-clazar: run up-to-date pipeline [2024-07-27] (#42617)

alafanechere committed:
🐙 source-goldcast: run up-to-date pipeline [2024-07-27] (#42616)

alafanechere committed:
🐙 source-google-directory: run up-to-date pipeline [2024-07-27] (#42615)

alafanechere committed:
🐙 source-gong: run up-to-date pipeline [2024-07-27] (#42614)

alafanechere committed:
🐙 source-mailjet-mail: run up-to-date pipeline [2024-07-27] (#42613)

alafanechere committed:
🐙 source-microsoft-dataverse: run up-to-date pipeline [2024-07-27] (#42612)

alafanechere committed:
🐙 source-ip2whois: run up-to-date pipeline [2024-07-27] (#42611)

alafanechere committed:
🐙 source-yotpo: run up-to-date pipeline [2024-07-27] (#42610)

alafanechere committed:
🐙 source-firebase-realtime-database: run up-to-date pipeline [2024-07-27] (#42609)

alafanechere committed:
🐙 source-linnworks: run up-to-date pipeline [2024-07-27] (#42608)

alafanechere committed:
🐙 source-vitally: run up-to-date pipeline [2024-07-27] (#42607)

alafanechere committed:
🐙 destination-typesense: run up-to-date pipeline [2024-07-27] (#42606)

alafanechere committed:
🐙 source-recruitee: run up-to-date pipeline [2024-07-27] (#42605)

alafanechere committed:
🐙 source-zendesk-sunshine: run up-to-date pipeline [2024-07-27] (#42604)

alafanechere committed:
🐙 source-pinterest: run up-to-date pipeline [2024-07-27] (#42603)

alafanechere committed:
🐙 source-surveycto: run up-to-date pipeline [2024-07-27] (#42602)

alafanechere committed:
🐙 source-gitlab: run up-to-date pipeline [2024-07-27] (#42601)

alafanechere committed:
🐙 source-launchdarkly: run up-to-date pipeline [2024-07-27] (#42600)

alafanechere committed:
🐙 source-coin-api: run up-to-date pipeline [2024-07-27] (#42599)

alafanechere committed:
🐙 source-unleash: run up-to-date pipeline [2024-07-27] (#42598)

alafanechere committed:
🐙 source-alpha-vantage: run up-to-date pipeline [2024-07-27] (#42597)

alafanechere committed:
🐙 source-statuspage: run up-to-date pipeline [2024-07-27] (#42596)

alafanechere committed:
🐙 source-us-census: run up-to-date pipeline [2024-07-27] (#42595)

alafanechere committed:
🐙 destination-pinecone: run up-to-date pipeline [2024-07-27] (#42594)

alafanechere committed:
🐙 source-twilio: run up-to-date pipeline [2024-07-27] (#42593)

alafanechere committed:
🐙 source-nasa: run up-to-date pipeline [2024-07-27] (#42592)

alafanechere committed:
🐙 source-ringcentral: run up-to-date pipeline [2024-07-27] (#42591)

alafanechere committed:
🐙 source-n8n: run up-to-date pipeline [2024-07-27] (#42590)

alafanechere committed:
🐙 source-chartmogul: run up-to-date pipeline [2024-07-27] (#42589)

alafanechere committed:
🐙 destination-databend: run up-to-date pipeline [2024-07-27] (#42588)

alafanechere committed:
🐙 source-mixpanel: run up-to-date pipeline [2024-07-27] (#42391)

alafanechere committed:
🐙 source-orbit: run up-to-date pipeline [2024-07-27] (#42388)

alafanechere committed:
🐙 source-appfollow: run up-to-date pipeline [2024-07-27] (#42387)

alafanechere committed:
🐙 source-survey-sparrow: run up-to-date pipeline [2024-07-27] (#42303)

alafanechere committed:
🐙 source-pipedrive: run up-to-date pipeline [2024-07-27] (#42287)

alafanechere committed:
🐙 source-yandex-metrica: run up-to-date pipeline [2024-07-27] (#42172)

alafanechere committed:
🐙 source-trustpilot: run up-to-date pipeline [2024-07-27] (#40267)

btkcodedev committed:
🐛Source XKCD: Fix metadata digest (#42834)

artem1205 committed:
fix(airbyte-cdk): fix declarative schema refs (#42844)

Signed-off-by: Artem Inzhyyants <artem.inzhyyants@gmail.com>

artem1205 committed:
🤖 patch bump Python CDK to version 3.9.6

artem1205 committed:
🤖 Cut version 3.9.6 of source-declarative-manifest

artem1205 committed:
perf(airbyte-cdk): performance enhancement (#42441)

Signed-off-by: Artem Inzhyyants <artem.inzhyyants@gmail.com>

artem1205 committed:
🤖 major bump Python CDK to version 4.0.0

darynaishchenko committed:
fix(source-intercom) raise config error on `Active subscription needed` error, transient error for Companies stream (#42094)

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
🐛 Source Shopify: fixed `store` redirection, add `bulk checkpointing` (#42095)

artem1205 committed:
feat(source-hardcoded-records): bump CDK version (#42850)

Signed-off-by: Artem Inzhyyants <artem.inzhyyants@gmail.com>

im-voracity committed:
✨ Source Instagram: add new fields for streams StoryInsights and MediaInsights (#41937)

Co-authored-by: Natik Gadzhi <natik@respawn.io>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

aldogonzalez8 committed:
🐛 bug(Airbyte CDK): fix case where stream wont have a state attribute and needs to resolve get_updated_state (#42083)

clnoll committed:
airbyte-ci: add validation before live test runs (#42463)

theyueli committed:
[cdk] Fix checkErrorType to not miss airbyte native config and transient error exceptions (#42839)

This bug causes airbyte ConfigErrorException and TransientErrorException to be rethrown and become system errors, although their error messages have been translated.

clnoll committed:
CATs: fix client config path (#42580)

aldogonzalez8 committed:
🤖 patch bump Python CDK to version 4.0.1

theyueli committed:
[source-postgres/mssql/mysql] use the latest CDK to use the latest fix of error translation (#42852)

AGPapa committed:
🐛 Source Braintre: Corrects pagination issues by adding page size to yml (#42848)

Commit:
[source-postgres] CAT test for CDC postgres (#42107)

Co-authored-by: Stephane Geneix <147216312+stephane-airbyte@users.noreply.github.com>
Co-authored-by: Catherine Noll <clnoll@users.noreply.github.com>
Co-authored-by: btkcodedev <btk.codedev@gmail.com>
Co-authored-by: Dhroov Makwana <pabloescoder@gmail.com>
Co-authored-by: Audrey Maldonado <audrey.maldonado@gmail.com>
Co-authored-by: Evan Tahler <evan@airbyte.io>
Co-authored-by: Yue Li <61070669+theyueli@users.noreply.github.com>
Co-authored-by: Augustin <augustin@airbyte.io>
Co-authored-by: Maxime Carbonneau-Leclerc <3360483+maxi297@users.noreply.github.com>
Co-authored-by: Alexandre Girard <alexandre@airbyte.io>
Co-authored-by: Baz <oleksandr.bazarnov@globallogic.com>
Co-authored-by: maxi297 <maxime@airbyte.io>
Co-authored-by: benmoriceau <benmoriceau@users.noreply.github.com>
Co-authored-by: Daryna Ishchenko <80129833+darynaishchenko@users.noreply.github.com>
Co-authored-by: Rodi Reich Zilberman <867491+rodireich@users.noreply.github.com>
Co-authored-by: Akash Kulkarni <113392464+akashkulk@users.noreply.github.com>
Co-authored-by: Edward Gao <edward.gao@airbyte.io>
Co-authored-by: Serhii Lazebnyi <53845333+lazebnyi@users.noreply.github.com>
Co-authored-by: Gireesh Sreepathi <gisripa@gmail.com>
Co-authored-by: Jeroen de Vries <jeroen@maildevries.eu>
Co-authored-by: prashant-mittal9 <168704883+prashant-mittal9@users.noreply.github.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>
Co-authored-by: lazebnyi <lazebnyi@users.noreply.github.com>
Co-authored-by: pmossman <pmossman@users.noreply.github.com>
Co-authored-by: Aldo Gonzalez <168454423+aldogonzalez8@users.noreply.github.com>
Co-authored-by: Davin Chia <davinchia@gmail.com>
Co-authored-by: Brian Lai <51336873+brianjlai@users.noreply.github.com>
Co-authored-by: Anton Karpets <karpets.as@gmail.com>
Co-authored-by: askarpets <anton.karpets@globallogic.com>
Co-authored-by: Serhii Lazebnyi <serhii.lazebnyi@globallogic.com>
Co-authored-by: Tim Werner <t.werner@comventure.de>
Co-authored-by: Natik Gadzhi <natik@respawn.io>
Co-authored-by: James Truty <jtruty@gmail.com>
Co-authored-by: Ben Church <ben@airbyte.io>
Co-authored-by: Marius Posta <marius@airbyte.io>
Co-authored-by: alafanechere <augustin.lafanechere@gmail.com>
Co-authored-by: Matias Samuel Miranda <matias.miranda@merantix-momentum.com>
Co-authored-by: Akash Kulkarni <akash@airbyte.io>

bgroff committed:
docs: fix doc problem with the secret key name. (#42554)

natikgadzhi committed:
docs(airbyte-cdk): clarified CHANGELOG entry on 4.0.0 python requirem… (#42870)

alafanechere committed:
airbyte-ci: send publish failures to a dedicated slack channel (#42849)

maxi297 committed:
feat(airbyte-cdk) Add ability to stop stream when retry-after is greater than a duration (#42865)

maxi297 committed:
🤖 patch bump Python CDK to version 4.0.2

evantahler committed:
Enterprise Connector Docs (#42564)

Co-authored-by: Alexandre Cuoci <alex.cuoci@airbyte.io>

clnoll committed:
airbyte-ci: add connectionOptions to metadata.yaml, for use by live tests (#42860)

natikgadzhi committed:
fix(source-declarative-manifest): allow Python 3.10 (#42871)

mr935h committed:
✨ Source Asana: add task and target data to stories compact stream (#42856)

bgroff committed:
Bump Airbyte version from 0.63.11 to 0.63.12

davinchia committed:
docs: worker v2 public docs. (#42873)

Update docs to prep for Worker V2 OSS launch.

The main focus is a brief explanation of what/why/how of the feature. We have a blog post in the works I'm going to link to for more explaination once it's published.

- Remove old *_WORKER configs.
- Update the diagram to reflect the new flow. Viewing now is not great due to the rendering. Will likely follow up on a different diagram.
- Write Worker V2 explanation. Note that I left all the Docker pieces in place and tried to make things 'flow'. Will follow up here when we deprecate Docker officially.

Co-authored-by: Jimmy Ma <gosusnp@users.noreply.github.com>

maxi297 committed:
🔧  Source Zendesk Support: Update BackoffStrategy interface (#42892)

bnchrch committed:
feat(cat): add manifest.yaml spec support (#42879)

Hesperide committed:
Enterprise Connector Viz Improvements + Workday Start (#42884)

davidmorera-rh committed:
✨ Source Paystack: add `reference` field to `Transfers` stream (#42901)

brianjlai committed:
🐛 [source-klaviyo] Add config option disable_fetching_predictive_analytics to prevent 503 Service Unavailable errors (#42895)

maxi297 committed:
feat(airbyte-cdk) Align BackoffStrategy interfaces to take attempt_count as a full-fled… (#42889)

maxi297 committed:
🤖 minor bump Python CDK to version 4.1.0

maxi297 committed:
🤖 Cut version 4.1.0 of source-declarative-manifest

alafanechere committed:
airbyte-ci: use a new cache volume for docker cache (#42905)

clnoll committed:
Update metadata.yaml for all sources with sandbox connections (#42577)

AGPapa committed:
feat(source-stripe): Expands discounts and tax rates in stripe invoice stream (from: @aasimsani) (#41985)

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

laskoviymishka committed:
🐙 source-dynamodb: fix incorrect type check for incremental read (#41936)

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

erohmensing committed:
auto-request reviewers for builder contributions (#42881)

clnoll committed:
CATs: do teardown outside of the session (#42874)

bgroff committed:
docs: add the authentication docs and update the quickstart to show how to get a password. (#42915)

Co-authored-by: Joey Marshment-Howell <josephkmh@users.noreply.github.com>

davinchia committed:
docs: add workloads blog post. (#42918)

lazebnyi commented on airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:
```diff
@@ -220,10 +220,48 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         return available_streams
 
+    # set of standard stream names to check for custom object name collisions
+    STANDARD_STREAM_NAMES = {
```
let's move this name to separate (e.g. utils) file and import after it there

lazebnyi commented on airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:
```diff
@@ -2245,9 +2245,10 @@ class CustomObject(CRMSearchStream, ABC):
+    def __init__(self, entity: str, name: str, schema: Mapping[str, Any], fully_qualified_name: str, custom_properties: Mapping[str, Any], **kwargs):
         super().__init__(**kwargs)
         self.entity = entity
+        self.name = name
```
we already have getter for `name` attribute, so better to proceed with this logic and add setter

jtruty replied:
I modified the name getter logic instead - this is much cleaner. Let me know what you think.

lazebnyi commented on airbyte-integrations/connectors/source-hubspot/unit_tests/test_streams.py:
```diff
@@ -407,6 +407,7 @@ def test_custom_object_stream_doesnt_call_hubspot_to_get_json_schema_if_availabl
 ):
     stream = CustomObject(
         entity="animals",
+        name="animals",
```
no need as `name` getter return `self.entity`

lazebnyi commented on airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:
```diff
@@ -220,10 +220,48 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         for entity, fully_qualified_name, schema, custom_properties in api.get_custom_objects_metadata():
             yield CustomObject(
                 entity=entity,
+                name=fully_qualified_name if entity in self.STANDARD_STREAM_NAMES else entity,
```
could you add a comment with examples for fully_qualified_name and entity to show the difference and provide more clarity?

jtruty replied:
Yes, done

jtruty committed:
Revert "Add support for Hubspot Deal Splits"

This reverts commit 20a532db7897b34c78b93b3a8385ef78364b3472.

jtruty committed:
pr feedback, simply logic

jtruty committed:
fix merge

jtruty committed:
fix pr

jtruty committed:
another pr fix

jtruty committed:
fix logic

natikgadzhi committed:
Merge branch 'master' into jtruty/handle_custom_object_name_collisions

natikgadzhi commented:
@jtruty thank you for this! Clean PR description, and strong work. @lazebnyi will be back on Monday, we'll take a look. Looks like this is close to wrapping up.

aldogonzalez8 commented:
@jtruty  Change seems straightforward, but can you add a unit test for this scenario? Probably "[airbyte-integrations/connectors/source-hubspot/unit_tests/test_streams.py](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/unit_tests/test_streams.py)" is a good place, if you a look to  [test_custom_object_stream_doesnt_call_hubspot_to_get_json_schema_if_available](https://github.com/airbytehq/airbyte/blob/c67c5ba4e3daf4ef4b8ad764cda002d7d36c4e39/airbyte-integrations/connectors/source-hubspot/unit_tests/test_streams.py#L405-L414) should be easy to add have one for your scenario.

Also, you have a conflict.

Thanks!
